Code/GraphMol/SubstructLibrary/SubstructLibrary.cpp

```diff
@@ -87,12 +88,30 @@ unsigned int SubstructLibrary::addMol(const ROMol &m) {
 }
 
 namespace {
+//  Return true if the pattern contains a ring query
+bool query_needs_rings(const ROMol &in_query) {
+  for (auto &atom: in_query.atoms()) {
+    if(atom->hasQuery()) {
+      if (describeQuery(atom).find("Ring") != std::string::npos) {
```

Comment:
Is there a better way?  this feels dirty :)

Reply:
There's not really